histFactory/CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ else()
 endif()
 
 set(Boost_NO_BOOST_CMAKE ON)
-find_package(Boost REQUIRED)
+find_package(Boost 1.57.0 REQUIRED COMPONENTS system filesystem)
```

Comment:
Do you specifically needs 1.57.0? It'd be better to not request any version if it works on older boost release (travis build is failing because of that)